examples/fedprox_example/client.py

```diff
@@ -52,7 +52,7 @@ def get_criterion(self, config: Config) -> _Loss:
     log(INFO, f"Device to be used: {DEVICE}")
     log(INFO, f"Server Address: {args.server_address}")
 
-    client = MnistFedProxClient(data_path, [Accuracy()], DEVICE)
+    client = MnistFedProxClient(data_path, [Accuracy()], DEVICE, use_wandb_reporter=True)
```

Comment:
Rather than having to set the `use_wandb_reporter` I think we can remove this boolean from the `BasicClient` and rely on the server config to send over the proper components. That is, in BasicClient, rather than doing 
```
if self.use_wandb_reporter:
    self.wandb_reporter = ClientWandBReporter.from_config(self.client_name, config)
```
we can just do 
```
self.wandb_reporter = ClientWandBReporter.from_config(self.client_name, config)
```
Since the `from_config` function handles creating from the passed config. We could simply augment the from config function to be more robust. That is, 
```
@classmethod
def from_config(cls, client_name: str, config: Dict[str, Any]) -> Optional["ClientWandBReporter"]:
        if "reporting_enabled" in config and config["reporting_enabled"]
            return ClientWandBReporter(client_name, config["project_name"], config["group_name"], config["entity"])
        else:
            return None
```

Reply:
Updated in [c9a023a](https://github.com/VectorInstitute/FL4Health/pull/61/commits/c9a023ab5d1f19dce69e969d263f81b47b112cc8)!